### PR TITLE
Refactoring in order to separate gospel to OCaml translation and assertion checking (default mode)

### PR DESCRIPTION
**Pull request by n-osborne**
As a follow up to the pluginisation of the monolith mode, we move the assertion checking to the default mode and expose the term translation function. This will allow the latter to be used in other modes (*eg* the qcheck-stm mode).

We use this opportunity to reshuffle functions into modules and rename these modules. So that it better matches their current usage.

co-authored by @shym

### CODE REVIEW

File: plugins/monolith/ortac-monolith.opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
+opam-version: "2.0"
+synopsis: "Monolith plugin got Ortac"
+description: "Monolith plugin got Ortac"
```
Comment (jmid) [inline plugins/monolith/ortac-monolith.opam:3]: Drive-by-comment: `for Ortac` (in both lines 2 and 3)

File: plugins/monolith/ortac-monolith.opam [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+maintainer: ["Nicolas Osborne <nicolas.osborne@tarides.cmd>"]
+authors: ["Nicolas Osborne <nicolas.osborne@tarides.cmd>"]
```
Comment (jmid) [inline plugins/monolith/ortac-monolith.opam:5]: It is `.com`, no (in lines 4 and 5)?

Comment (shym) [reply plugins/monolith/ortac-monolith.opam:5]: Indeed. It was fixed in #69 but missed in some rebasing of this PR it seems ;-)

Comment (n-osborne) [reply plugins/monolith/ortac-monolith.opam]: Fixed :-)

File: plugins/monolith/ortac-monolith.opam (hunk 1 above)
Comment (n-osborne) [reply plugins/monolith/ortac-monolith.opam]: Fixed. Thanks :-)
